docs/source/dev/py_postprocessing.rst

```diff
@@ -0,0 +1,178 @@
+.. _development-pytools:
+
+Python postprocessing tool structure
+====================================
+
+For each plugin there are (or have to be implemented) three different classes.
+
+1. A data reader class responsible for loading the data from the simulation directory
+2. A visualizer class that outputs a matplotlib plot
+3. A jupyter-widget class that exposes the parameters of the matplotlib visualizer to the user
+   via other widgets
+
```

Comment:
The repository directory for PIConGPU Python modules for plugins is ``lib/python/picongpu/plugins/``.